src/components/portal/portal.tsx

```diff
@@ -11,9 +11,9 @@ import { FlipModifier } from '@popperjs/core/lib/modifiers/flip';
  *
  * * Events might not bubble up as expected since the content is moved out to another DOM node
  * * Any styling that is applied to content from the parent will be lost, if the content is
- *   just another web compoent it will work without any issues
- * * When the node is moved in the DOM, `componentDidUnload`, `disconnectedCallback` and `connectedCallback`
- *   will be invoked, so `componentDidUnload` can not be used as a destructor (which is the wrong behavior anyway)
+ * just another web compoent it will work without any issues
+ * * When the node is moved in the DOM, `disconnectedCallback` and `connectedCallback`
+ * will be invoked, so `disconnectedCallback` can not be used as a destructor (which is the wrong behavior anyway)
```

Comment:
disconnectedCallback

We might have to look into this comment, as `disconnectedCallback` should probably actually be used as a destructor 😄

Reply:
The Stencil documentation says that you should not use it as a destructor
> Called every time the component is disconnected from the DOM, ie, it can be dispatched more than once, DO not confuse with a "onDestroy" kind of event.

Reply:
Yeah, but is there something else to use then? We can't just leak instead… 😕

Reply:
In the example, they certainly seem to clean up in `disconnectedCallback`…
https://stenciljs.com/docs/component-lifecycle#example

I think perhaps they just mean "don't assume the component will be gone forever after this". Or that comment was just forgotten and left in the docs when `componentDidUnload` was deprecated…

Reply:
Yeah I think you can do cleanup in `disconnectedCallback`, but you have to assume that the component might be added to the DOM again and initialize everything again